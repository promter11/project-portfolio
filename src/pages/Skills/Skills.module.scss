@import "../../styles/_variables";

.skills {
  height: 100%;
  padding: 8rem 14rem;
  background: rgba(0, 0, 0, 0.75) url("../../assets/images/Skills.jpg")
    no-repeat fixed center center / cover;
  background-blend-mode: darken;
}

.title {
  margin-bottom: 8rem;
  text-align: right;
  letter-spacing: 1rem;
  opacity: 0.25;
}

.content {
  margin-bottom: 8rem;
}

.contentTitle {
  position: relative;
  margin-bottom: 9rem;
  font-size: 2.4rem;
  font-weight: 900;
  line-height: 2.8rem;
  color: $white-color;

  &:before {
    content: "";
    position: absolute;
    bottom: -1rem;
    width: 4rem;
    height: 0.4rem;
    background-color: $white-color;
  }
}

.wrapper {
  display: flex;
  justify-content: space-between;
}

.block {
  text-align: center;
  color: $white-color;
}

.image {
  width: 6rem;
  height: auto;
  margin-bottom: 2rem;
}

.blockTitle {
  margin-bottom: 2rem;
  font-size: 1.8rem;
  font-weight: 900;
  line-height: 1;
  text-transform: uppercase;
}

.text {
  max-width: 18rem;
  margin: 0 auto;
  font-size: 1.4rem;
  font-weight: 100;
  line-height: 150%;
}

.desc {
  font-size: 1.4rem;
  font-weight: 100;
  line-height: 1;
  color: $white-color;
}
