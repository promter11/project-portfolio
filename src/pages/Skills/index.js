import React from "react";

import HTML from "../../assets/icons/skills/html-5.svg";
import CSS from "../../assets/icons/skills/css-3.svg";
import JavaScript from "../../assets/icons/skills/javascript.svg";
import styles from "./Skills.module.scss";

const initialState = [
  {
    id: 0,
    title: "HTML",
    text: "Lorem ipsum dolor sit amet, ipsum dolor sit amet",
    image: HTML,
  },
  { id: 1, title: "CSS", text: "Lorem ipsum dolor sit amet", image: CSS },
  {
    id: 2,
    title: "JavaScript",
    text: "Lorem ipsum dolor sit amet",
    image: JavaScript,
  },
];

const Skills = () => {
  return (
    <section className={styles.skills}>
      <h1 className={styles.title}>Skills</h1>
      <div className={styles.content}>
        <h2 className={styles.contentTitle}>Basics</h2>
        <div className={styles.wrapper}>
          {initialState.map(({ id, title, text, image }, _) => {
            return (
              <div key={id} className={styles.block}>
                <img className={styles.image} src={image} alt="Technology" />
                <h3 className={styles.blockTitle}>{title}</h3>
                <p className={styles.text}>{text}</p>
              </div>
            );
          })}
        </div>
      </div>
      <p className={styles.desc}>
        * And other popular technologies like React Router, Styled Components,
        Redux Form and others.
      </p>
    </section>
  );
};

export default Skills;
