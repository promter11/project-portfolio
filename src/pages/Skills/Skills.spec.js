import React from "react";
import { shallow } from "enzyme";

import Skills from "./index";

const props = {
  language: "RU",
};

describe("Skills page", () => {
  it("renders correctly with props", () => {
    const wrapper = shallow(<Skills {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
