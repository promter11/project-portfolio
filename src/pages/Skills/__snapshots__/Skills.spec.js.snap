// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skills page renders correctly with props 1`] = `
<section
  className="skills"
>
  <Container>
    <Heading
      className="title"
      level={1}
    >
      Навыки
    </Heading>
    <div
      className="content"
    >
      <div
        className="contentBlock"
        key="0"
      >
        <Heading
          className="contentTitle"
          level={2}
        >
          Структура документа
        </Heading>
        <div
          className="wrapper"
        >
          <div
            className="block"
            key="0"
          >
            <img
              alt="HTML"
              className="image"
              src="html.svg"
            />
            <Heading
              className="blockTitle"
              level={3}
            >
              HTML
            </Heading>
            <p
              className="text"
            >
              Язык разметки для создания структуры веб-страниц и электронных писем
            </p>
          </div>
          <div
            className="block"
            key="1"
          >
            <img
              alt="Pug"
              className="image"
              src="pug.svg"
            />
            <Heading
              className="blockTitle"
              level={3}
            >
              Pug
            </Heading>
            <p
              className="text"
            >
              Препроцессор HTML и шаблонизатор
            </p>
          </div>
        </div>
      </div>
      <div
        className="contentBlock"
        key="1"
      >
        <Heading
          className="contentTitle"
          level={2}
        >
          Внешний вид документа
        </Heading>
        <div
          className="wrapper"
        >
          <div
            className="block"
            key="0"
          >
            <img
              alt="CSS"
              className="image"
              src="css.svg"
            />
            <Heading
              className="blockTitle"
              level={3}
            >
              CSS
            </Heading>
            <p
              className="text"
            >
              Формальный язык, используемый для описания внешнего вида документа
            </p>
          </div>
          <div
            className="block"
            key="1"
          >
            <img
              alt="Sass"
              className="image"
              src="sass.svg"
            />
            <Heading
              className="blockTitle"
              level={3}
            >
              Sass
            </Heading>
            <p
              className="text"
            >
              Метаязык на основе CSS, предназначенный для повышения уровня абстракции кода CSS и упрощения файлов CSS
            </p>
          </div>
          <div
            className="block"
            key="2"
          >
            <img
              alt="Bootstrap"
              className="image"
              src="bootstrap.svg"
            />
            <Heading
              className="blockTitle"
              level={3}
            >
              Bootstrap
            </Heading>
            <p
              className="text"
            >
              Фреймворк HTML / CSS / JS для быстрого создания современных адаптивных веб-сайтов
            </p>
          </div>
          <div
            className="block"
            key="3"
          >
            <img
              alt="БЭМ"
              className="image"
              src="bem.svg"
            />
            <Heading
              className="blockTitle"
              level={3}
            >
              БЭМ
            </Heading>
            <p
              className="text"
            >
              Методология веб-разработки, а также набор интерфейсных библиотек, фреймворков и вспомогательных инструментов
            </p>
          </div>
        </div>
      </div>
      <div
        className="contentBlock"
        key="2"
      >
        <Heading
          className="contentTitle"
          level={2}
        >
          Языки программирования
        </Heading>
        <div
          className="wrapper"
        >
          <div
            className="block"
            key="0"
          >
            <img
              alt="JavaScript"
              className="image"
              src="javascript.svg"
            />
            <Heading
              className="blockTitle"
              level={3}
            >
              JavaScript
            </Heading>
            <p
              className="text"
            >
              Динамический язык программирования, который применяется к HTML-документу и может обеспечивать динамическую интерактивность на веб-сайтах
            </p>
          </div>
          <div
            className="block"
            key="1"
          >
            <img
              alt="TypeScript"
              className="image"
              src="typescript.svg"
            />
            <Heading
              className="blockTitle"
              level={3}
            >
              TypeScript
            </Heading>
            <p
              className="text"
            >
              Строго типизированный язык программирования, который добавляет синтаксические преимущества языку Javascript
            </p>
          </div>
          <div
            className="block"
            key="2"
          >
            <img
              alt="PHP"
              className="image"
              src="php.svg"
            />
            <Heading
              className="blockTitle"
              level={3}
            >
              PHP
            </Heading>
            <p
              className="text"
            >
              Скриптовый, интерпретируемый язык программирования, специально разработанный для написания веб-приложений (скриптов), запускаемых на веб-сервере
            </p>
          </div>
        </div>
      </div>
      <div
        className="contentBlock"
        key="3"
      >
        <Heading
          className="contentTitle"
          level={2}
        >
          Библиотеки
        </Heading>
        <div
          className="wrapper"
        >
          <div
            className="block"
            key="0"
          >
            <img
              alt="jQuery"
              className="image"
              src="jquery.svg"
            />
            <Heading
              className="blockTitle"
              level={3}
            >
              jQuery
            </Heading>
            <p
              className="text"
            >
              Быстрая, небольшая и многофункциональная библиотека JavaScript
            </p>
          </div>
          <div
            className="block"
            key="1"
          >
            <img
              alt="React"
              className="image"
              src="react.svg"
            />
            <Heading
              className="blockTitle"
              level={3}
            >
              React
            </Heading>
            <p
              className="text"
            >
              Библиотека JavaScript для создания пользовательских интерфейсов
            </p>
          </div>
          <div
            className="block"
            key="2"
          >
            <img
              alt="Redux"
              className="image"
              src="redux.svg"
            />
            <Heading
              className="blockTitle"
              level={3}
            >
              Redux
            </Heading>
            <p
              className="text"
            >
              Библиотека для управления состоянием приложения
            </p>
          </div>
          <div
            className="block"
            key="3"
          >
            <img
              alt="MobX"
              className="image"
              src="mobx.svg"
            />
            <Heading
              className="blockTitle"
              level={3}
            >
              MobX
            </Heading>
            <p
              className="text"
            >
              Простое масштабируемое управление состоянием
            </p>
          </div>
          <div
            className="block"
            key="4"
          >
            <img
              alt="Material UI"
              className="image"
              src="material.svg"
            />
            <Heading
              className="blockTitle"
              level={3}
            >
              Material UI
            </Heading>
            <p
              className="text"
            >
              Компоненты React для более быстрой и простой веб-разработки
            </p>
          </div>
        </div>
      </div>
      <div
        className="contentBlock"
        key="4"
      >
        <Heading
          className="contentTitle"
          level={2}
        >
          Инструменты
        </Heading>
        <div
          className="wrapper"
        >
          <div
            className="block"
            key="0"
          >
            <img
              alt="NPM"
              className="image"
              src="npm.svg"
            />
            <Heading
              className="blockTitle"
              level={3}
            >
              NPM
            </Heading>
            <p
              className="text"
            >
              Менеджер пакетов, включенный в Node.js
            </p>
          </div>
          <div
            className="block"
            key="1"
          >
            <img
              alt="Gulp"
              className="image"
              src="gulp.svg"
            />
            <Heading
              className="blockTitle"
              level={3}
            >
              Gulp
            </Heading>
            <p
              className="text"
            >
              Набор инструментов для автоматизации и улучшения рабочего процесса
            </p>
          </div>
          <div
            className="block"
            key="2"
          >
            <img
              alt="Webpack"
              className="image"
              src="webpack.svg"
            />
            <Heading
              className="blockTitle"
              level={3}
            >
              Webpack
            </Heading>
            <p
              className="text"
            >
              Сборщик модулей
            </p>
          </div>
        </div>
      </div>
      <div
        className="contentBlock"
        key="5"
      >
        <Heading
          className="contentTitle"
          level={2}
        >
          Другое
        </Heading>
        <div
          className="wrapper"
        >
          <div
            className="block"
            key="0"
          >
            <img
              alt="Git"
              className="image"
              src="git.svg"
            />
            <Heading
              className="blockTitle"
              level={3}
            >
              Git
            </Heading>
            <p
              className="text"
            >
              Бесплатная распределенная система контроля версий с открытым исходным кодом
            </p>
          </div>
          <div
            className="block"
            key="1"
          >
            <img
              alt="MySQL"
              className="image"
              src="mysql.svg"
            />
            <Heading
              className="blockTitle"
              level={3}
            >
              MySQL
            </Heading>
            <p
              className="text"
            >
              Бесплатная система управления реляционными базами данных
            </p>
          </div>
        </div>
      </div>
    </div>
    <p
      className="desc"
    >
      * И другие популярные технологии, такие как React Router, Styled Components, Redux ContactForm, React Transition Group и другие.
    </p>
  </Container>
</section>
`;
